src/compiler/scala/tools/nsc/backend/jvm/BTypesFromSymbols.scala

```diff
@@ -761,6 +761,6 @@ abstract class BTypesFromSymbols[G <: Global](val global: G) extends BTypes {
     javaFlags(sym) |
       ( if (sym hasAnnotation TransientAttr) asm.Opcodes.ACC_TRANSIENT else 0) |
       ( if (sym hasAnnotation VolatileAttr)  asm.Opcodes.ACC_VOLATILE  else 0) |
-      ( if (sym.isMutable) 0 else asm.Opcodes.ACC_FINAL)
+      ( if (sym.isMutable || sym.hasAllFlags(symtab.Flags.notPRIVATE)) 0 else asm.Opcodes.ACC_FINAL)
```

Comment:
There's a risk that this affects other fields. Could we instead add the `mutable` flag in the specialization phase?

Reply:
Interesting idea - I will try it out 👍

Reply:
Scratch the remark about `isEffectivelyFinal` - it only applies to inlining (and hence methods)